### PR TITLE
Add the `Remove-PoshGitFromProfile` function

### PR DESCRIPTION
**Pull request by dscho**
This is needed to be able to install/uninstall Posh-Git via Git for Windows' installer/uninstaller.

See https://github.com/git-for-windows/build-extra/pull/401 for details.

### CODE REVIEW

**Comment by dscho** [comment]
Please note that the CI failures are independent of the changes introduced by this here PR. At least [some of them](https://app.travis-ci.com/github/dahlbyk/posh-git/builds/243248233#L395-L562) seem to have been [reported before](https://github.com/dahlbyk/posh-git/pull/864#issuecomment-875378865).

**Comment by dscho** [comment]
@dahlbyk ping?

**Comment by dscho** [comment]
Sorry to be a bother, but if we want to offer Posh-Git as an option in Git for Windows v2.35.0, we kinda need to move a _little_ faster ;-)

File: src/Utils.ps1 [hunk 1]
```diff
@@ -216,6 +216,123 @@ function Add-PoshGitToProfile {
...
+        $underTest = $true
+        if ($TestParams.Count -gt 1) {
+            $ModuleBasePath = [string]$TestParams[1]
+        }
+    }
+
+    if (!$profilePath) { $profilePath = $PROFILE }
+
+    if (!$profilePath) {
+        Write-Warning "Skipping add of posh-git import to profile; no profile found."
```
Comment (dahlbyk) [inline src/Utils.ps1:290]:
```suggestion
        Write-Warning "Skipping removal of posh-git import from profile; no profile found."
```

File: src/Utils.ps1 [hunk 2]
```diff
@@ -216,6 +216,123 @@ function Add-PoshGitToProfile {
...
+        # If the profile script exists and is signed, then we should not modify it
+        if (!(Get-Command Get-AuthenticodeSignature -ErrorAction SilentlyContinue))
+        {
+            Write-Verbose "Platform doesn't support script signing, skipping test for signed profile."
+        }
+        else {
+            $sig = Get-AuthenticodeSignature $profilePath
+            if ($null -ne $sig.SignerCertificate) {
+                Write-Warning "Skipping add of posh-git import to profile; '$profilePath' appears to be signed."
+                Write-Warning "Add the command 'Import-Module posh-git' to your profile and resign it."
```
Comment (dahlbyk) [inline src/Utils.ps1:309]:
```suggestion
                Write-Warning "Skipping removal of posh-git import from profile; '$profilePath' appears to be signed."
                Write-Warning "Remove the command 'Import-Module posh-git' from your profile and resign it."
```

File: src/Utils.ps1 [hunk 3]
```diff
@@ -216,6 +216,123 @@ function Add-PoshGitToProfile {
...
+
+            if($line -like '. *posh-git*profile.example.ps1*') {
+                continue;
+            }
+            if($line -like 'Import-Module *\posh-git.psd1*') {
+                continue;
+            }
+            $newProfile += $line
+        }
+        Set-Content -path $profilePath -value $newProfile -Force -Encoding $oldProfileEncoding
```
Comment (dahlbyk) [inline src/Utils.ps1:332]: Noting this is pulled from: https://github.com/dahlbyk/posh-git/blob/dc2b6367324a9db79db55186375462a454f29c2e/chocolatey/tools/chocolateyUninstall.ps1#L7-L25

Will be nice to use this function there.

**Pull request merged**